client/context/context.go

```diff
@@ -51,17 +52,18 @@ type CLIContext struct {
 	FromName      string
 	Indent        bool
 	SkipConfirm   bool
+	SecretStore   bool
 }
 
 // NewCLIContextWithFrom returns a new initialized CLIContext with parameters from the
 // command line using Viper. It takes a key name or address and populates the FromName and
 // FromAddress field accordingly.
-func NewCLIContextWithFrom(from string) CLIContext {
+func NewCLIContextWithFrom(from string, input io.Reader) CLIContext {
```

Comment:
Why do we need an `io.Reader` here? We should rather not break the existing API(s) and require an `io.Reader` just to get an address.

If it's a must, then the `io.Reader` should already exist or be set on the context.

Reply:
We need to pass the return value of the call `cmd.InOrStdin()`, I see the point of introducing this change.
Though I agree with @alexanderbez, ideally this API should not change and an input buffer should be passed via a `func (ctx CLIContext) SetInput(io.Reader) Context` method.

Reply:
The problem is that if the input is not passed in, it will cause problems when the new keyring secret store gets initiated. The input is required for the keyring secret store because we need the input to pass in the password. The password input unlocks the keyring store.  The way it is implemented seems to be the best way.  Let me know your thoughts. Thanks!

Reply:
Sure, I understand the need for the `io.Reader`, but what I'm suggesting is:

```go
context.NewCLIContext().WithReader(reader).WithFromFields(fromAddrStr)
```

Reply:
Okay I understand the proposed refactor

Reply:
See updates to the API